pandas/_libs/index.pyx

```diff
@@ -26,6 +26,18 @@ from pandas._libs import algos, hashtable as _hash
 from pandas._libs.tslibs import Timestamp, Timedelta, period as periodlib
 from pandas._libs.missing import checknull
 
+# Python 2 vs Python 3
+try:
+    from thread import allocate_lock as _thread_allocate_lock
+except:
+    try:
+        from _thread import allocate_lock as _thread_allocate_lock
+    except:
+        try:
+            from dummy_thread import allocate_lock as _thread_allocate_lock
+        except:
+            from _dummy_thread import allocate_lock as _thread_allocate_lock
+
```

Comment:
Is it possible to move this into `pandas/compat/__init__.py` and import from there?

Reply:
why not just import it from tslibs.strptime where it is already?

Reply:
In that case, we should move that import code to `pandas/compat` so it can be utilized by both modules (in more semantically logical arrangement IMO).

Reply:
Can you address this? We also recently fixed all our bare excepts.

Reply:
That sounds fine, I didn't read through all the comments :)